### PR TITLE
Add support for constant variable

### PR DESCRIPTION
**Pull request by quoctruong**
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-dotnet-debugger/337)

### CODE REVIEW

File: src/google_cloud_debugger/google_cloud_debugger_lib/dbg_stack_frame.h [hunk 1]
```diff
@@ -239,6 +240,17 @@ class DbgStackFrame : public IDbgStackFrame {
       const std::vector<google_cloud_debugger_portable_pdb::LocalVariableInfo>
           &variable_infos);
 
+  // Parses the local constant from constant_infos.
+  HRESULT ProcessLocalConstants(
+      const std::vector<google_cloud_debugger_portable_pdb::LocalConstantInfo>
+          &constant_infos);
+
+  // Parses local constant that is an enum.
+  HRESULT ProcessEnumConstant(const std::string &constant_name,
```
Comment (iantalarico) [inline src/google_cloud_debugger/google_cloud_debugger_lib/dbg_stack_frame.h]: nit maybe ProcessLocalEnumConstant to be explicit?

File: src/google_cloud_debugger/google_cloud_debugger_lib/dbg_stack_frame.cc [hunk 2]
```diff
@@ -194,6 +204,119 @@ HRESULT DbgStackFrame::ProcessLocalVariables(
...
+      encoded_token = *((uint16_t *)enum_metadata_buffer.data());
+      break;
+    case 4:
+      encoded_token = *((uint32_t *)enum_metadata_buffer.data());
+      break;
+    case 8:
+      encoded_token = *((uint64_t *)enum_metadata_buffer.data());
+      break;
+    default:
+      invalid_token = true;
```
Comment (iantalarico) [inline src/google_cloud_debugger/google_cloud_debugger_lib/dbg_stack_frame.cc]: Can't you just return a failure here?

File: src/google_cloud_debugger/google_cloud_debugger_lib/cor_debug_helper.cc [hunk 3]
```diff
@@ -1106,4 +1106,77 @@ HRESULT CorDebugHelper::PopulateGenericClassTypesFromClassObject(
...
+  // divisible by sizeof(WCHAR) and get the length of the string.
+  if (*cor_type == CorElementType::ELEMENT_TYPE_STRING) {
+    if ((signature_blob.size() - 1) % sizeof(WCHAR) != 0) {
+      cerr << "Not enough bytes to read string value for constant.";
+      return E_FAIL;
+    }
+    *value_len = (signature_blob.size() - 1) / sizeof(WCHAR);
+  }
+
+  *constant_value = (UVCP_CONSTANT)(signature_blob.data() + 1);
```
[the remark this answers is not in the document]
Comment (quoctruong) [reply src/google_cloud_debugger/google_cloud_debugger_lib/cor_debug_helper.cc]: Added comment.

**Comment by quoctruong** [comment]
@iantalarico PTAL. I will fix and add tests in the PR after this.

**Comment by quoctruong** [comment]
Thanks!

**Pull request merged**